### PR TITLE
nrf_wifi: Add reset statistics command and api

### PR DESCRIPTION
**Pull request by kapbh**
Add api and command to reset the statistics of fw. NRF_WIFI_CMD_RESET_STATISTICS will be send to UMAC. Which will reset the statistics.

### CODE REVIEW

File: nrf_wifi/fw_if/umac_if/src/fmac_api_common.c [hunk 1]
```diff
@@ -1301,3 +1301,34 @@ enum nrf_wifi_status nrf_wifi_fmac_set_packet_filter(void *dev_ctx, unsigned cha
...
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		goto out;
+	}
+
+	do {
+		nrf_wifi_osal_sleep_ms(fmac_dev_ctx->fpriv->opriv,
+				       1);
+		count++;
+	} while ((fmac_dev_ctx->stats_req == true) &&
+		 (count < NRF_WIFI_FMAC_STATS_RECV_TIMEOUT));
```
Comment (krish2718) [inline nrf_wifi/fw_if/umac_if/src/fmac_api_common.c:1321]:
```suggestion
	} while ((fmac_dev_ctx->stats_req == true) &&
		 (count++ < NRF_WIFI_FMAC_STATS_RECV_TIMEOUT));
```

File: nrf_wifi/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 2]
```diff
@@ -1603,6 +1605,16 @@ struct nrf_wifi_event_deinit_done {
 	struct nrf_wifi_sys_head sys_head;
 } __NRF_WIFI_PKD;
 
+/**
+ * @brief This structure describes command for reset interface statistics.
+ *
+ */
+
```
Comment (krish2718) [inline nrf_wifi/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1612]: extra newline

File: nrf_wifi/fw_if/umac_if/src/fmac_api_common.c [hunk 3]
```diff
@@ -1301,3 +1301,33 @@ enum nrf_wifi_status nrf_wifi_fmac_set_packet_filter(void *dev_ctx, unsigned cha
...
 }
 #endif /* CONFIG_NRF700X_RAW_DATA_RX || CONFIG_NRF700X_PROMISC_DATA_RX */
+
+enum nrf_wifi_status nrf_wifi_fmac_stats_reset(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	unsigned char count = 0;
+
+	status = umac_cmd_prog_stats_reset(fmac_dev_ctx);
+
```
Comment (krish2718) [inline nrf_wifi/fw_if/umac_if/src/fmac_api_common.c:1311]: can be removed

**Review by krish2718 — APPROVED** [review]

File: nrf_wifi/fw_if/umac_if/inc/fmac_api_common.h [hunk 4]
```diff
@@ -362,6 +362,17 @@ enum nrf_wifi_status nrf_wifi_fmac_set_packet_filter(void *dev_ctx, unsigned cha
 						     unsigned short buffer_size);
 #endif /* CONFIG_NRF700X_RAW_DATA_RX || CONFIG_NRF700X_PROMISC_DATA_RX */
 
+/**
+ * @brief Issue a request to reset stats of the RPU.
+ * @param fmac_dev_ctx Pointer to the UMAC IF context for a RPU WLAN device.
+ *
+ * This function is used to send a command to
+ *          instruct the firmware to reset current RPU statistics.
```
Comment (sachinthegreen) [inline nrf_wifi/fw_if/umac_if/inc/fmac_api_common.h:370]: Why blank space before `instruct` ?

File: nrf_wifi/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 5]
```diff
@@ -1603,6 +1605,16 @@ struct nrf_wifi_event_deinit_done {
 	struct nrf_wifi_sys_head sys_head;
 } __NRF_WIFI_PKD;
 
+/**
+ * @brief This structure describes command for reset interface statistics.
```
Comment (sachinthegreen) [inline nrf_wifi/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1609]:
```suggestion
 * @brief This structure describes the command for reset of interface statistics.
```

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: nrf_wifi/fw_if/umac_if/inc/fmac_api_common.h (hunk 4 above)
Comment (kapbh) [reply nrf_wifi/fw_if/umac_if/inc/fmac_api_common.h]: I have seen similar format in file. Removed now.

**Review by sachinthegreen — APPROVED** [review]

**Pull request merged**